operator/backupcontroller/executor.go

```diff
@@ -42,54 +46,122 @@ func (b *BackupExecutor) Execute(ctx context.Context) error {
 	return b.startBackup(ctx)
 }
 
+type backupItem struct {
+	volume      corev1.Volume
+	node        string
+	tolerations []corev1.Toleration
+	targetPod   string
+}
+
 // listAndFilterPVCs lists all PVCs in the given namespace and filters them for K8up specific usage.
 // Specifically, non-RWX PVCs will be skipped, as well PVCs that have the given annotation.
-func (b *BackupExecutor) listAndFilterPVCs(ctx context.Context, annotation string) ([]corev1.Volume, error) {
+func (b *BackupExecutor) listAndFilterPVCs(ctx context.Context, annotation string) ([]backupItem, error) {
 	log := controllerruntime.LoggerFrom(ctx)
-	volumes := make([]corev1.Volume, 0)
+
+	pods := &corev1.PodList{}
+	pvcPodMap := make(map[string]corev1.Pod)
+	if err := b.Config.Client.List(ctx, pods, client.InNamespace(b.backup.Namespace)); err != nil {
+		return nil, fmt.Errorf("list pods: %w", err)
+	}
+	for _, pod := range pods.Items {
+		for _, volume := range pod.Spec.Volumes {
+			if volume.PersistentVolumeClaim != nil {
+				pvcPodMap[volume.PersistentVolumeClaim.ClaimName] = pod
+				log.V(1).Info("pvc pod map", "claimName", volume.PersistentVolumeClaim.ClaimName, "pod", pod.GetName())
+			}
+		}
+	}
+
+	backupItems := make([]backupItem, 0)
 	claimlist := &corev1.PersistentVolumeClaimList{}
 
 	log.Info("Listing all PVCs", "annotation", annotation)
 	if err := b.fetchPVCs(ctx, claimlist); err != nil {
-		return volumes, err
+		return backupItems, err
 	}
 
-	for _, item := range claimlist.Items {
-		annotations := item.GetAnnotations()
+	for _, pvc := range claimlist.Items {
+		if pvc.Status.Phase != corev1.ClaimBound {
+			log.Info("PVC is not bound", "pvc", pvc.GetName())
+			continue
+		}
 
-		tmpAnnotation, ok := annotations[annotation]
+		backupAnnotation, hasBackupAnnotation := pvc.GetAnnotations()[annotation]
 
-		if !containsAccessMode(item.Spec.AccessModes, "ReadWriteMany") && !ok {
-			log.Info("PVC isn't RWX", "pvc", item.GetName())
+		isRWO := containsAccessMode(pvc.Spec.AccessModes, corev1.ReadWriteOnce)
+		if !containsAccessMode(pvc.Spec.AccessModes, corev1.ReadWriteMany) && !isRWO && !hasBackupAnnotation {
+			log.Info("PVC is neither RWX nor RWO and has no backup annotation", "pvc", pvc.GetName())
 			continue
 		}
 
-		if !ok {
-			log.Info("PVC doesn't have annotation, adding to list", "pvc", item.GetName())
-		} else if anno, _ := strconv.ParseBool(tmpAnnotation); !anno {
-			log.Info("PVC skipped due to annotation", "pvc", item.GetName(), "annotation", tmpAnnotation)
+		if !hasBackupAnnotation {
```

Comment:
Do I see that right, that this basically inverts the functionality of the annotation?

E.g. Instead of annotate to include, we should annotate to exclude?

If so we should probably treat this as a breaking change.

Reply:
Yeah, you're right, I didn't see the negation in the old version :)